File: Sources/EwECustomPlugins/EwEValueChain/Data/Flow elements/Diagrams/cFlowPosition.vb
' SPDX-License-Identifier: EUPL-1.2
' This file is part of Ecopath with Ecosim (EwE).
' Copyright © 1991– Ecopath International Initiative (EII)

Imports EwECore.Database.cEwEDatabase

''' ===========================================================================
''' <summary>
''' Position of a single unit in a flow diagram.
''' </summary>
''' ===========================================================================
Public Class cFlowPosition
    Inherits cOOPStorable

#Region " Private vars "

    Private m_diagram As cFlowDiagram = Nothing
    Private m_unit As cUnit = Nothing

    Private m_iX As Integer = 0
    Private m_iY As Integer = 0
    Private m_iWidth As Integer = 0
    Private m_iHeight As Integer = 0

#End Region ' Private vars

#Region " Properties "

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the diagram this flow position belongs to.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Public Property Diagram() As cFlowDiagram
        Get
            Return Me.m_diagram
        End Get
        Set(value As cFlowDiagram)
            If (Not ReferenceEquals(value, Me.m_diagram)) Then
                Me.m_diagram = value
                Me.SetChanged()
            End If
        End Set
    End Property

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the unit that this position belongs to.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Public Property Unit() As cUnit
        Get
            Return Me.m_unit
        End Get
        Set(value As cUnit)
            If (Not ReferenceEquals(value, Me.m_unit)) Then
                Me.m_unit = value
                Me.SetChanged()
            End If
        End Set
    End Property

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the X position.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Property Xpos() As Integer
        Get
            Return Me.m_iX
        End Get
        Set(value As Integer)
            If (value <> Me.m_iX) Then
                Me.m_iX = value
                Me.SetChanged()
            End If
        End Set
    End Property

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the Y position.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Property Ypos() As Integer
        Get
            Return Me.m_iY
        End Get
        Set(value As Integer)
            If (value <> Me.m_iY) Then
                Me.m_iY = value
                Me.SetChanged()
            End If
        End Set
    End Property

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the width.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Property Width() As Integer
        Get
            Return Me.m_iWidth
        End Get
        Set(value As Integer)
            If (value <> Me.m_iWidth) Then
                Me.m_iWidth = value
                Me.SetChanged()
            End If
        End Set
    End Property

    ''' -----------------------------------------------------------------------
    ''' <summary>
    ''' Get/set the height.
    ''' </summary>
    ''' -----------------------------------------------------------------------
    Property Height() As Integer
        Get
            Return Me.m_iHeight
        End Get
        Set(value As Integer)
            If (value <> Me.m_iHeight) Then
                Me.m_iHeight = value
                Me.SetChanged()
            End If
        End Set
    End Property

#End Region ' Properties

End Class

File: Sources Internal/ValueChain/ValueChain/Data/Flow elements/Diagrams/cFlowPosition.cs
// ===============================================================================
// This file is part of Ecopath with Ecosim (EwE)
//
// EwE is free software: you can redistribute it and/or modify it under the terms
// of the GNU General Public License version 2 as published by the Free Software 
// Foundation.
//
// EwE is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR 
// PURPOSE. See the GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License along with EwE.
// If not, see <http://www.gnu.org/licenses/gpl-2.0.html>. 
//
//
// Copyright 1991- 
//    Ecopath International Initiative, Barcelona, Spain
// ===============================================================================

#region  Imports 


namespace ValueChain
{

    #endregion

    /// ===========================================================================
/// <summary>
/// Position of a single unit in a flow diagram.
/// </summary>
/// ===========================================================================
    public class cFlowPosition : cValueChainEntity
    {

        #region  Private vars 

        private cFlowDiagram m_diagram = null;
        private cUnit m_unit = null;

        private int m_iX = 0;
        private int m_iY = 0;
        private int m_iWidth = 0;
        private int m_iHeight = 0;

        #endregion

        #region  Properties 

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the diagram this flow position belongs to.
    /// </summary>
    /// -----------------------------------------------------------------------
        public cFlowDiagram Diagram
        {
            get
            {
                return m_diagram;
            }
            set
            {
                if (!ReferenceEquals(value, m_diagram))
                {
                    m_diagram = value;
                    SetChanged();
                }
            }
        }

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the unit that this position belongs to.
    /// </summary>
    /// -----------------------------------------------------------------------
        public cUnit Unit
        {
            get
            {
                return m_unit;
            }
            set
            {
                if (!ReferenceEquals(value, m_unit))
                {
                    m_unit = value;
                    SetChanged();
                }
            }
        }

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the X position.
    /// </summary>
    /// -----------------------------------------------------------------------
        public int Xpos
        {
            get
            {
                return m_iX;
            }
            set
            {
                if (value != m_iX)
                {
                    m_iX = value;
                    SetChanged();
                }
            }
        }

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the Y position.
    /// </summary>
    /// -----------------------------------------------------------------------
        public int Ypos
        {
            get
            {
                return m_iY;
            }
            set
            {
                if (value != m_iY)
                {
                    m_iY = value;
                    SetChanged();
                }
            }
        }

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the width.
    /// </summary>
    /// -----------------------------------------------------------------------
        public int Width
        {
            get
            {
                return m_iWidth;
            }
            set
            {
                if (value != m_iWidth)
                {
                    m_iWidth = value;
                    SetChanged();
                }
            }
        }

        /// -----------------------------------------------------------------------
    /// <summary>
    /// Get/set the height.
    /// </summary>
    /// -----------------------------------------------------------------------
        public int Height
        {
            get
            {
                return m_iHeight;
            }
            set
            {
                if (value != m_iHeight)
                {
                    m_iHeight = value;
                    SetChanged();
                }
            }
        }

        #endregion

    }
}